### PR TITLE
Upstream 1.2.1

### PR DESCRIPTION
**Pull request by erichelgeson**
* Reduce delay between phase signal changes
* Reinsert CD-ROM image when receiving Inquiry command - off by default
* Open image as read-only if attribute set - merged with existing read-only cdrom feature.

### CODE REVIEW

File: src/BlueSCSI_disk.cpp [hunk 1]
```diff
@@ -356,10 +356,11 @@ class ImageBackingStore
         }
         else
         {
-            if(type == S2S_CFG_OPTICAL)
+            m_isreadonly_attr = !!(FS_ATTRIB_READ_ONLY & SD.attrib(filename));
```
Comment (androda) [inline src/BlueSCSI_disk.cpp:359]: Is the "not not" here doing anything useful?

File: src/BlueSCSI_disk.cpp [hunk 2]
```diff
@@ -544,7 +550,7 @@ class ImageBackingStore
 
     void flush()
     {
-        if (!m_israw && !m_isrom)
+        if (!m_israw && !m_isrom && !m_isreadonly_attr)
```
Comment (androda) [inline src/BlueSCSI_disk.cpp:553]: Could also be expressed this way
`if (! (m_israw || m_isrom || m_isreadonly_attr))`

File: src/BlueSCSI_disk.cpp [hunk 3]
```diff
@@ -2113,6 +2149,16 @@ void scsiDiskPoll()
             image_config_t &img = *(image_config_t*)scsiDev.target->cfg;
             checkDiskGeometryDivisible(img);
         }
+
+        // Check for Inquiry command to reinsert CD-ROMs on boot
+        if (command == 0x12)
+        {
+            image_config_t &img = *(image_config_t*)scsiDev.target->cfg;
+            if (img.deviceType == S2S_CFG_OPTICAL && img.reinsert_on_inquiry)
```
Comment (androda) [inline src/BlueSCSI_disk.cpp:2157]: Smaller perf penalty would be to change reinsert on inquiry first in the if block

**Review by androda — APPROVED** [review]
Overall looks fine, plus nitpicky comments

File: src/BlueSCSI_disk.cpp (hunk 1 above)
Comment (erichelgeson) [reply src/BlueSCSI_disk.cpp:359]: Converting an int to a c++ bool it seems https://stackoverflow.com/questions/6764168/what-does-double-not-used-on-a-non-boolean-variable-do

**Pull request merged**